ngraph/src/ngraph/op/non_max_suppression.cpp

```diff
@@ -349,18 +353,22 @@ void op::v3::NonMaxSuppression::validate_and_infer_types()
                           "Expected a scalar for the 'max_output_boxes_per_class' input. Got: ",
                           max_boxes_ps);
 
-    const auto iou_threshold_ps = get_input_partial_shape(3);
-    NODE_VALIDATION_CHECK(this,
-                          iou_threshold_ps.is_dynamic() || is_scalar(iou_threshold_ps.to_shape()),
-                          "Expected a scalar for the 'iou_threshold' input. Got: ",
-                          iou_threshold_ps);
+    if (get_inputs().size() >= 4) {
```

Comment:
Inside inference-engine/src/transformations/src/transformations/convert_opset1_to_legacy/convert_nms_to_nms_ie.cpp
we take 4th and 5th inputs without any condition. Also NonMaxSuppressionIE operation requires 5 inputs. So my question is how it works? Let's add nms to nms ie test for case with optional inputs.
Also according to spec opset1::NonMaxSuppression has only two required inputs, so it's better to add the same check for 3rd input.

Reply:
The clone_with_new_inputs always create all 5 inputs. That is how it works.